examples/eth_dns_support.yml

```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- hosts: all
+  vars:
+    network_connections:
+      - name: eth0
+        type: ethernet
+        ip:
+          route_metric4: 100
+          dhcp4: no
+          #dhcp4_send_hostname: no
+          gateway4: 192.0.2.1
+
+          dns:
+            - 192.0.2.2
+            - 198.51.100.5
+          dns_search:
+            - example.com
+            - subdomain.example.com
+          dns_options:
+            - rotate
+            - timeout:1
```

Comment:
How about:

```yaml
dns:
  servers:
    - 1.1.1.1
    - 8.8.8.8
  searches:
    - example.com
    - example.org
   options:
    - free_string_a
```

Reply:
`dns:` and `dns_search:` already exist, only `dns_options:` was added, so this would be a bigger change. Another thing: Using plural `searches` sounds odd to me.

Reply:
@tyll , @cathay4t , Indeed, I already finished implementation for grouping dns into what Gris suggested, but I found a lot of test failures in  './tests/unit/test_network_connections.py'.

Reply:
As `dns:` and `dns_search` already exist, I am OK to place `dns_options` along with them.